ui-tests/src/main/java/org/example/pages/HomePage.java

```diff
@@ -0,0 +1,42 @@
+package org.example.pages;
+
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Condition.*;
+import static com.codeborne.selenide.Selectors.byId;
+import static com.codeborne.selenide.Selenide.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.openqa.selenium.By.className;
+import static org.openqa.selenium.By.cssSelector;
+
+public class HomePage {
+
+    private static final By BANNER = className("img-fluid");
+    private static final By QUERY_FIELD = byId("search-query");
+    private static final By SEARCH_BTN = cssSelector(".input-group .btn-secondary");
+    private static final By CARDS_COLLECTION = className("card-title");
+    private static final By CARD_BODY = className("card-img-top");
+    private static final By PRODUCT_PRICE_HP = cssSelector(".float-end.text-muted [data-test='product-price']");
+    private String productId = "01J26MFEZ4T4M6HFCRFRQCXTJB";
+
+    public HomePage isLoaded() {
+        $(BANNER).shouldBe(visible);
+        return this;
+    }
+
+    public String searchProductByName(String productName) {
+        $(QUERY_FIELD).val(productName);
+        $(SEARCH_BTN).shouldBe(clickable).click();
+        String title = $$(CARDS_COLLECTION).first().shouldHave(text(productName)).text();
+        assertThat(title).isEqualTo(productName);
+        String priceText = $$(PRODUCT_PRICE_HP).first().shouldBe(visible).text();
+        return priceText.replace("$", "").trim();
+    }
+
+    public void goOnProductPage(){
+        $(CARD_BODY).shouldBe(clickable).click();
+        String currentUrl = webdriver().driver().url();
+        assertThat(currentUrl).contains(productId);
```

Comment:
Move this assertion to assertion class, it is incorrect to put assertions to the page object.
Exception could be for methods like isLoaded